ReactAndroid/src/main/java/com/facebook/react/devsupport/MultipartStreamReader.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.devsupport;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import okio.Buffer;
+import okio.BufferedSource;
+import okio.ByteString;
+
+/**
+ * Utility class to parse the body of a response of type multipart/mixed.
+ */
+public class MultipartStreamReader {
+  private static final String CRLF = "\r\n";
+
+  private final BufferedSource mSource;
+  private final String mBoundary;
+
+  public interface ChunkCallback {
+    void execute(Map<String, String> headers, Buffer body, boolean done) throws IOException;
+  }
+
+  public MultipartStreamReader(BufferedSource source, String boundary) {
+    mSource = source;
+    mBoundary = boundary;
+  }
+
+  private Map<String, String> parseHeaders(Buffer data) {
+    Map<String, String> headers = new HashMap<>();
+
+    String text = data.readUtf8();
+    String[] lines = text.split(CRLF);
```

Comment:
If the line separator `"\r\n"` also when running the packager on Mac and Linux? If yes, can you add a comment explaining why this is always the line separator on all platforms?